dll/appcompat/shims/layer/forcedxsetupsuccess.c

```diff
@@ -0,0 +1,268 @@
+/*
+ * PROJECT:     ReactOS 'Layers' Shim library
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     ForceDxSetupSuccess shim
+ * COPYRIGHT:   Copyright 2019 Mark Jansen (mark.jansen@reactos.org)
+ */
+
+#define WIN32_NO_STATUS
+#include <windef.h>
+#include <winbase.h>
+#include <shimlib.h>
+#include "ntndk.h"
+
+typedef HMODULE(WINAPI* LOADLIBRARYAPROC)(LPCSTR lpLibFileName);
+typedef HMODULE(WINAPI* LOADLIBRARYWPROC)(LPCWSTR lpLibFileName);
+typedef HMODULE(WINAPI* LOADLIBRARYEXAPROC)(LPCSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+typedef HMODULE(WINAPI* LOADLIBRARYEXWPROC)(LPCWSTR lpLibFileName, HANDLE hFile, DWORD dwFlags);
+typedef FARPROC(WINAPI* GETPROCADDRESSPROC)(HMODULE hModule, LPCSTR lpProcName);
+typedef BOOL   (WINAPI* FREELIBRARYPROC)(HINSTANCE hLibModule);
+
+
+#define SHIM_NS         ForceDxSetupSuccess
+#include <setup_shim.inl>
+
+
+#define DSETUPERR_SUCCESS 0
+
+INT WINAPI DirectXSetup(HWND hWnd, LPSTR lpszRootPath, DWORD dwFlags)
+{
+    SHIM_MSG("Returning DSETUPERR_SUCCESS\n");
+    return DSETUPERR_SUCCESS;
+}
+
+INT WINAPI DirectXSetupA(HWND hWnd, LPSTR lpszRootPath, DWORD dwFlags)
+{
+    SHIM_MSG("Returning DSETUPERR_SUCCESS\n");
+    return DSETUPERR_SUCCESS;
+}
+
+INT WINAPI DirectXSetupW(HWND hWnd, LPWSTR lpszRootPath, DWORD dwFlags)
+{
+    SHIM_MSG("Returning DSETUPERR_SUCCESS\n");
+    return DSETUPERR_SUCCESS;
+}
+
+INT WINAPI DirectXSetupGetVersion(DWORD *lpdwVersion, DWORD *lpdwMinorVersion)
+{
+    if (lpdwVersion)
+        *lpdwVersion = MAKELONG(7, 4);     // DirectX 7.0
+    if (lpdwMinorVersion)
+        *lpdwMinorVersion = MAKELONG(1792, 0);
+
+    return TRUE;
+}
+
+
+static
+BOOLEAN
+IsCharInAnsiString(
+    IN CHAR Char,
+    IN const STRING* MatchString)
+{
+    USHORT i;
+
+    for (i = 0; i < MatchString->Length; i++)
+    {
+        if (Char == MatchString->Buffer[i])
+            return TRUE;
+    }
+
+    return FALSE;
+}
+
+
+NTSTATUS
+NTAPI
+FindCharInAnsiString(
+    IN const STRING* SearchString,
+    IN const STRING* MatchString,
+    OUT PUSHORT Position)
+{
+    BOOLEAN Found;
+    ULONG i, Length;
+
+    *Position = 0;
+
+    /* Search */
+    Length = SearchString->Length;
+    for (i = Length - 1; (LONG)i >= 0; i--)
+    {
+        Found = IsCharInAnsiString(SearchString->Buffer[i], MatchString);
+        if (Found)
+        {
+            *Position = i;
+            return STATUS_SUCCESS;
+        }
+    }
+
+    return STATUS_NOT_FOUND;
+}
+
+
+static BOOL IsDxSetupA(const PSTRING LibraryPath)
```

Comment:
Should this just do AnsiStringToUnicodeString and then call the -W version or is there a need to duplicate it?

Reply:
Thought about that, but wanted to save on an allocation.
Might be easier / better maintainable.